### PR TITLE
Swagger fix for product EI

### PR DESCRIPTION
**Pull request by milindaperera**
## Purpose
Add CORS support for APIs

This fix provide CORS support for APIs for as a part of adding  swagger editor support for APIs
Fix: wso2/product-ei#4113

Add publishSwagger support for API

Fix: wso2/product-ei#3328
This fix will add ability to publish swagger definition for synapse API

### CODE REVIEW

File: modules/core/src/main/java/org/apache/synapse/mediators/builtin/RespondMediator.java [hunk 1]
```diff
@@ -28,6 +31,14 @@ public boolean mediate(MessageContext synCtx) {
 
         synCtx.setTo(null);
         synCtx.setResponse(true);
+
+        // if this is not a response from a proxy service
+        String proxyName = (String) synCtx.getProperty(SynapseConstants.PROXY_SERVICE);
+        if (proxyName == null || "".equals(proxyName)) {
```
Comment (dulanjalidilmi) [inline modules/core/src/main/java/org/apache/synapse/mediators/builtin/RespondMediator.java]: Shall we use isEmpty() method to check empty string?

Comment (sameeragunarathne) [inline modules/core/src/main/java/org/apache/synapse/mediators/builtin/RespondMediator.java]: can we use StringUtils.isEmpty() here?

Comment (milindaperera) [reply modules/core/src/main/java/org/apache/synapse/mediators/builtin/RespondMediator.java]: Fixed, thanks

Comment (milindaperera) [reply modules/core/src/main/java/org/apache/synapse/mediators/builtin/RespondMediator.java]: Fixed, thanks

**Pull request merged**